calendarsubscription/http.go

```diff
@@ -0,0 +1,51 @@
+package calendarsubscription
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/target/goalert/permission"
+	"github.com/target/goalert/util/errutil"
+)
+
+// ServeICalData will return an iCal file for the subscription associated with the current request.
+func (b *Store) ServeICalData(w http.ResponseWriter, req *http.Request) {
+	ctx := req.Context()
+	src := permission.Source(ctx)
+	if src.Type != permission.SourceTypeCalendarSubscription {
+		http.Error(w, http.StatusText(http.StatusForbidden), http.StatusForbidden)
+		return
+	}
+
+	cs, err := b.FindOne(ctx, src.ID)
+	if errutil.HTTPError(ctx, w, err) {
+		return
+	}
+
+	var n time.Time
+	err = b.now.QueryRowContext(ctx).Scan(&n)
+	if errutil.HTTPError(ctx, w, err) {
+		return
+	}
+
+	shifts, err := b.oc.HistoryBySchedule(ctx, cs.ScheduleID, n, n.AddDate(0, 1, 0))
+	if errutil.HTTPError(ctx, w, err) {
+		return
+	}
+
+	// filter out other users
+	filtered := shifts[:0]
+	for _, s := range shifts {
+		if s.UserID != cs.UserID {
+			continue
+		}
+		filtered = append(filtered, s)
```

Comment:
As a general rule in Go code (and in js), we try to keep main operations flat, short-circuit "error" conditions, and end with the normal flow at the end of the block like this.

You'll always see the filter conditions first (and also clean to add more) and then the append or other operation at the end.

```go
// instead of 
for {
  if cond1 && cond2 {
    // successful flow
  }
  // success + fail
}

// do
for {
  if !cond1 {
    // fail
    continue
  }
  if !cond2 {
    // fail
    continue
  }
  // successful flow only
}

// or
for {
  if !cond1 || !cond2 {
    // fail
    continue
  }
  // successful flow only
}
```

https://golang.org/doc/effective_go.html#if
> The code reads well if the successful flow of control runs down the page, eliminating error cases as they arise. Since error cases tend to end in return statements, the resulting code needs no else statements.